GiveNGo/GiveNGo
# Chat Feed Setup

## Description:
- New inclusions to the application allow users to navigate from the main chat page to a separate feed page. The addition of a ChatRouter file makes this possible. When a user selects a given chat from the ChatPage, this will route users to a feed page to converse with other users regarding unfinished tasks. Additionally, the inclusion of Websockets allows for a more dynamic method of communication when users are within the ChatFeed page. 

## Changes I Made:
- Setup chat navigation logic  inside the **ChatRouter** file to navigate between the **ChatPage and ChatFeed** screens
- Updated **Footer.tsx** to include newly created ChatRouter file
- Created styling and submission logic for new chats typed in the **ChatFeed** page
- Incorporated Web-sockets into the application with the inclusion of a **Sockets.js** file
- Changed ChatFeed **View component** to allow list of scrollable **chat Bubble** components

## How to Test:
- run `npm install`
- run `npm start`
- Navigate from the **Login Page** to the **Chat Page**
- Select one of the **open chats** and this will route you to the **ChatFeed** page
- Once there, enter some text into the **InputField** and **click submit**
- Your message should display on the feed page

![Screen Shot 2020-07-23 at 11 40 16 AM](https://user-images.githubusercontent.com/55554784/88324950-4b35ad80-ccd9-11ea-861f-8a77d1116441.png)
